test/integration/vaults/vault-creation.integration.ts

```diff
@@ -0,0 +1,246 @@
+    validatorExitRequesters: HardhatEthersSigner,
+    validatorWithdrawalTriggerers: HardhatEthersSigner,
+    disconnecters: HardhatEthersSigner,
+    nodeOperatorFeeClaimers: HardhatEthersSigner,
+    stranger: HardhatEthersSigner;
+
+  let allRoles: HardhatEthersSigner[];
+  let snapshot: string;
+  let originalSnapshot: string;
+
+  before(async () => {
+    ctx = await getProtocolContext();
+
+    originalSnapshot = await Snapshot.take();
+
+    const { depositSecurityModule, stakingVaultFactory } = ctx.contracts;
+    await depositSecurityModule.DEPOSIT_CONTRACT();
+
+    allRoles = await getRandomSigners(20);
+    [
+      owner,
+      nodeOperatorManager,
+      funder,
+      withdrawer,
+      minter,
+      burner,
+      assetRecoverer,
+      rebalancer,
+      depositPausers,
+      depositResumers,
+      validatorExitRequesters,
+      validatorWithdrawalTriggerers,
+      disconnecters,
+      nodeOperatorFeeClaimers,
+      stranger,
+    ] = allRoles;
+    console.log(allRoles.map((s) => s.address));
+    // Owner can create a vault with operator as a node operator
+    const deployTx = await stakingVaultFactory.connect(owner).createVaultWithDelegation(
+      {
+        defaultAdmin: owner,
+        nodeOperatorManager: nodeOperatorManager,
+        assetRecoverer: assetRecoverer,
+        nodeOperatorFeeBP: VAULT_NODE_OPERATOR_FEE,
+        confirmExpiry: days(7n),
+        funders: [funder],
+        withdrawers: [withdrawer],
+        minters: [minter],
+        burners: [burner],
+        rebalancers: [rebalancer],
+        depositPausers: [depositPausers],
+        depositResumers: [depositResumers],
+        validatorExitRequesters: [validatorExitRequesters],
+        validatorWithdrawalTriggerers: [validatorWithdrawalTriggerers],
+        disconnecters: [disconnecters],
+        nodeOperatorFeeClaimers: [nodeOperatorFeeClaimers],
+      },
+      "0x",
+    );
+    const createVaultTxReceipt = (await deployTx.wait()) as ContractTransactionReceipt;
+    const createVaultEvents = ctx.getEvents(createVaultTxReceipt, "VaultCreated");
+
+    expect(createVaultEvents.length).to.equal(1n);
+
+    stakingVault = await ethers.getContractAt("StakingVault", createVaultEvents[0].args?.vault);
+    delegation = await ethers.getContractAt("Delegation", createVaultEvents[0].args?.owner);
+    await setupLido(ctx);
+    // only equivalent of 10.0% of TVL can be minted as stETH on the vault
+  });
+
+  beforeEach(async () => (snapshot = await Snapshot.take()));
+
+  afterEach(async () => await Snapshot.restore(snapshot));
+
+  after(async () => await Snapshot.restore(originalSnapshot));
+
+  async function generateFeesToClaim() {
+    const { vaultHub } = ctx.contracts;
+    const hubSigner = await impersonate(await vaultHub.getAddress(), ether("100"));
+    const rewards = ether("1");
+    await stakingVault.connect(hubSigner).report(rewards, 0n, 0n);
+  }
+
+  it("Allows to fund an withdraw funds for dedicated roles", async () => {
```

Comment:
Maybe add contexts like in the doc? Vault owner + subroles, NO + subroles, anyone... would make review easier. wdyt?